File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>

using namespace std;

int main(int argc, char** argv)
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    queue<int> Q;
    for (int i = 0; i < n; i++) {
        string command;
        cin >> command;
        if (command == "push") {
            int input;
            cin >> input;
            Q.push(input);
        }
        else if (command == "pop") {
            if (Q.empty()) cout << "-1";
            else {
                cout << Q.front();
                Q.pop();
            }
            cout << "\n";
        }
        else if (command == "size") {
            cout << Q.size();
            cout << "\n";
        }
        else if (command == "empty") {
            if (Q.empty()) cout << "1";
            else cout << "0";
            cout << "\n";
        }
        else if (command == "front") {
            if (Q.empty()) cout << "-1";
            else {
                cout << Q.front();
            }
            cout << "\n";
        }
        else if (command == "back") {
            if (Q.empty()) cout << "-1";
            else {
                cout << Q.back();
            }
            cout << "\n";

        }
    }

    return 0;
}

File: 백준/Silver/10845. 큐/큐.py
from collections import deque

import sys
input = sys.stdin.readline

n = int(input())
dq = deque()
for i in range(n):
    order = input().split()
    if order[0] == "push":
        dq.append(order[1])
    elif order[0] == "pop":
        if len(dq) == 0:
            print(-1)
        else:
            print(dq.popleft())
    elif order[0] == "size":
        print(len(dq))
    elif order[0] == "empty":
        if len(dq) == 0:
            print(1)
        else:
            print(0)
    elif order[0] == "front":
        if len(dq) == 0:
            print(-1)
        else:
            print(dq[0])
    elif order[0] == "back":
        if len(dq) == 0:
            print(-1)
        else:
            print(dq[len(dq)-1])